### DISCUSSION TITLE
react-native-mapbox-gl pricing

### DISCUSSION THREAD

**Discussion by sauhardar** [rnmapbox/maps#1216] [Q&A] [answered]
Hey! Does anyone know how pricing works with this library? Does this library use the mobile SDKs under the hood? If so, does that mean that Mapbox would charge according to their Maps SDKs for Mobile rate? Thanks.

**Comment by ferdicus** [comment, marked as answer]
Hi @sauhardar, 
yes this repo is basically a wrapper around the native iOS and Android SDKs. 

Therefore _their_ rules apply. 

You might've noticed, that you also have to set up and use a mapbox API key when setting up this lib and using map tiles from mapbox, that is, because by default this will render a mapbox map. 

Having all that said: 
You can direct the `MapView` to point to a different tile server all together - hence not hitting mapboxs servers. 

Then you don't have to pay anything. 

We're making sure, that the default pack-in SDK version still adhere to that. 

We can't guarantee that for the newer versions though - so carefully read their licensing docs before you apply an update

**Reply by sauhardar** [reply]
Are there any docs for how I can get `MapView` to point to a different tile server?

**Reply by ferdicus** [reply]
There are various ways, I'd recommend you check the `/example` folder for some pointers.
You can either render raster tiles, via passing `RasterSource`/ `RasterLayer` - there you could point to a tile server of your choice or check out this [example](https://github.com/react-native-mapbox-gl/maps/blob/master/example/src/examples/Map/ShowMapLocalStyle.tsx) which utilizes a local `tile.json` file. That can point to wherever you get vector maps from. 
Can be your own hosted map tiles server, or mapbox again with some style changes or another third party, like HERE for example.
